### PR TITLE
BICAWS-1873: Enable processing results from Bichard to be compared

### PR DESCRIPTION
**Pull request by tomvaughan77**
[BICAWS-1873](https://dsdmoj.atlassian.net/browse/BICAWS-1873). Add a script to compare and summarize the output of old Bichard against new. Currently this just compares the `triggers` and `exceptions` to check the same codes are present.

Changes:
* Added `scripts/compareResults.ts` to compare output of old Bichard against new. This can be run with `npm run compare:dev` locally (enables pretty printing), and `npm run compare` in production environments.
* Added `Comparing New and Old Bichard` section to README
* Tweaked zod schema in `src/types/IncomingMessage.ts` for `ConvictingCourt` to fix obvious comparison error.
* Changed all imports in repo to be relative (`../src/...` instead of `src/...`) to fix script. This is annoying and makes up the VAST bulk of this PR (45/50 files changed).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+Old Bichard is currently configured to publish its outputs (AHO, triggers, exceptions), plus the original message it received to the `PROCESSING_VALIDATION_QUEUE` queue in ActiveMQ. We can compare these outputs against new Bichard by running:
+
+```bash
+npm i # If packages not already installed
+npm run compare
+```
+
+This requires the [old Bichard](https://github.com/ministryofjustice/bichard7-next) stack to be running. Outputs from old Bichard can be driven onto the queue by running either `make pushq` in the [old Bichard repo](https://github.com/ministryofjustice/bichard7-next), or by running `npm t` in the [e2e testing repo](https://github.com/ministryofjustice/bichard7-next-tests).
+
+This will run the `scripts/compareResults.ts` script to check the output of old Bichard against new Bichard. A tally is kept of the results and can be seen by exiting the script with `Ctrl-C`. If `compareResults.ts` can't process any messages off the queue (empty messages, ActiveMQ errors ect), these are recorded as `skipped` and will be counted seperatly.
```
Comment (jaskaransarkaria) [inline README.md:32]: Can you make a little more explicit that it's the npm run compare cmd that runs `compareResults.ts`. My first reading of it led me to think that the previous para's cmds ran the script

**Review by bjpirt — APPROVED** [review]

File: scripts/compareResults.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+import type BichardResultType from "../src/types/BichardResultType"
+
+interface BichardResult {
+  incomingMessage: string
+  annotatedHearingOutcome: string
+  triggers?: Trigger[]
+  exceptions?: Exception[]
+}
+
+const logger: Pino.Logger = Pino({
```
Comment (jaskaransarkaria) [inline scripts/compareResults.ts:19]: I think it would be nice if this was in it's own file (so it could be reused) and then imported here.

**Review by jaskaransarkaria — APPROVED** [review]
LGTM 🎉

File: README.md (hunk 1 above)
Comment (tomvaughan77) [reply README.md]: Yep! Done :)

File: scripts/compareResults.ts (hunk 2 above)
Comment (tomvaughan77) [reply scripts/compareResults.ts]: Good point. It now lives under `src/utils/logging.ts` :)

**Review by jaskaransarkaria — APPROVED** [review]

**Review by jaskaransarkaria — APPROVED** [review]

**Pull request merged**